fiftyone/core/collections.py

```diff
@@ -2809,6 +2928,48 @@ def set_label_values(
                 progress=progress,
             )
 
+    def _get_sample_ids_for_labels(self, label_field, label_ids):
+        is_frame_field = self._is_frame_field(label_field)
+        root, is_list_field = self._get_label_field_root(label_field)
+        label_id_path = root + ".id"
+
+        # We only need `view` to contain labels we actually want to
+        # process, so if the number of values is small enough that
+        # `select_labels()` may optimize, we use it
+        if len(label_ids) <= 100000:
+            view = self.select_labels(ids=label_ids, fields=label_field)
+        else:
+            view = self
+
+        id_map = {}
+
+        if is_frame_field:
+            _frame_ids, _label_ids = view.values(["frames.id", label_id_path])
+
+            if is_list_field:
+                for _fids, _flids in zip(_frame_ids, _label_ids):
+                    for _frame_id, _lids in zip(_fids, _flids):
+                        if _lids:
+                            for _label_id in _lids:
+                                id_map[_label_id] = _frame_id
+            else:
+                for _fids, _flids in zip(_frame_ids, _label_ids):
+                    for _frame_id, _label_id in zip(_fids, _flids):
+                        id_map[_label_id] = _frame_id
+        else:
+            _sample_ids, _label_ids = view.values(["id", label_id_path])
+
+            if is_list_field:
+                for _sample_id, _lids in zip(_sample_ids, _label_ids):
+                    if _lids:
+                        for _label_id in _lids:
+                            id_map[_label_id] = _sample_id
+            else:
+                for _sample_id, _label_id in zip(_sample_ids, _label_ids):
+                    id_map[_label_id] = _sample_id
```

Comment:
How does this compare to composing dict and zip?
```suggestion
                id_map = dict(zip(_label_ids, _sample_ids))
```